### PR TITLE
Add demo to README

### PR DESCRIPTION
**Pull request by romilbhardwaj**
This PR updates the example in README.md to run mnist training from the [pytorch examples](https://github.com/pytorch/examples). Also adds a gif demo for the example.

### CODE REVIEW

**Comment by WoosukKwon** [comment]
It looks good! BTW, why don't we set `SKYPILOT_MINIMIZE_LOGGING=1` to make the logging messages more concise? I think we did so in the last demo.

**Comment by romilbhardwaj** [comment]
> It looks good! BTW, why don't we set `SKYPILOT_MINIMIZE_LOGGING=1` to make the logging messages more concise? I think we did so in the last demo.

That's a great point! I agree, it definitely looks cleaner with minimized logging. However, we had a brief discussion and decided maybe its good to keep the demo in the repo true to what a user would see on their first run.

**Comment by WoosukKwon** [comment]
@romilbhardwaj Got it. I think that also makes sense.

**Review by Michaelvll — APPROVED** [review]
Thanks for adding the gif @romilbhardwaj ! LGTM!

File: README.md [hunk 1]
```diff
@@ -39,23 +39,28 @@ num_nodes: 1
...
 # Typical use: make use of resources, such as running training.
 run: |
-  echo "Hello, SkyPilot!"
-  conda env list
+  cd mnist
+  python main.py --epochs 1
 ```
 
-This task can be launched on the cloud with the `sky launch` command.
+For this example, you can prepare the workdir by cloning the pytorch examples repository.
```
Comment (concretevitamin) [inline README.md:56]: I'd push for a minimal amount of words here. Something like:
```
Prepare the wordir by cloning:

Clone the repo to be used as workdir:
```

File: README.md [hunk 2]
```diff
@@ -39,23 +39,28 @@ num_nodes: 1
...
 ```
 
-This task can be launched on the cloud with the `sky launch` command.
+For this example, you can prepare the workdir by cloning the pytorch examples repository. 
 ```bash
-$ sky launch my-task.yaml
+git clone https://github.com/pytorch/examples.git ~/torch_examples
+```
+
+This task can now be launched on the cloud with the `sky launch` command.
```
Comment (concretevitamin) [inline README.md:61]: Likewise:
```
Launch with `sky launch`:
```

**Review by concretevitamin — APPROVED** [review]
Thanks @romilbhardwaj - this is great!

**Pull request merged**